### PR TITLE
Moved Steps to kite-extras

### PR DESCRIPTION
**Pull request by ManuCosmo**
Pass 'StepParams' object to BasePage and TestStep constructor to pass a single object instead of stepPhase, Logger, webdriver, and more to come...

### CODE REVIEW

File: src/main/java/io/cosmosoftware/kite/pages/WebRTCInternalPage.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+   * @return the webrtc-internals dump
+   * @throws KiteInteractionException
+   */
+  public String downloadDump() throws KiteInteractionException {
+    if (!isChrome(webDriver)) {
+      return "webrtc-internals is only supported on Chrome";
+    }
+    ((JavascriptExecutor) webDriver).executeScript("window.open()");
+    ArrayList<String> tabs = new ArrayList<>(webDriver.getWindowHandles());
+    webDriver.switchTo().window(tabs.get(1));
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/pages/WebRTCInternalPage.java]: Might create IndexOutOfBound exeption

File: src/main/java/io/cosmosoftware/kite/steps/ScreenshotStep.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  public ScreenshotStep(StepParams params) {
+    super(params);
+    setStepPhase(StepPhase.ALL);
+  }
+
+  @Override
+  protected void step() throws KiteTestException {
+    Reporter.getInstance()
+      .screenshotAttachment(
+        report, "ScreenshotStep_" + timestamp(), saveScreenshotPNG(webDriver));
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/steps/ScreenshotStep.java]: There's an exception BrowserCantBeReached or something. THis happens sometimes when taking screenshot, and get thrown as UnHanddled Exception. Might need to check to see whether this is caught in saveScreenshotPNG()

File: src/main/java/io/cosmosoftware/kite/util/WebDriverUtils.java [hunk 3]
```diff
@@ -383,6 +383,16 @@ public static boolean isElectron(WebDriver webDriver) {
     Capabilities capabilities = ((RemoteWebDriver) webDriver).getCapabilities();
     return "chrome".equals(capabilities.getBrowserName()) && capabilities.getVersion().isEmpty();
   }
+
+  /**
+   * @return true if it is chrome
+   */
+  public static boolean isChrome(WebDriver webDriver) {
+    Capabilities capabilities = ((RemoteWebDriver) webDriver).getCapabilities();
+    return "chrome".equals(capabilities.getBrowserName());
```
Comment (namvuCosmo) [inline src/main/java/io/cosmosoftware/kite/util/WebDriverUtils.java]: this is also true for eclectron based app

Comment (ManuCosmo) [reply src/main/java/io/cosmosoftware/kite/util/WebDriverUtils.java]: it's fine. what I want to do on Chrome should also work on electron (i guess)

File: src/main/java/io/cosmosoftware/kite/steps/ScreenshotStep.java (hunk 2 above)
Comment (ManuCosmo) [reply src/main/java/io/cosmosoftware/kite/steps/ScreenshotStep.java]: that should result in a broken step. I didnt encounter this so far.

**Pull request merged**